compiler/rustc_borrowck/src/diagnostics/move_errors.rs

```diff
@@ -291,6 +296,12 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
         self.buffer_error(err);
     }
 
+    fn maybe_copy(&mut self, ty: Ty<'tcx>) -> bool {
+        use rustc_trait_selection::infer::InferCtxtExt;
+        let Some(copy_trait_def) = self.infcx.tcx.lang_items().copy_trait() else { return false };
+        self.infcx.type_implements_trait(copy_trait_def, [ty], self.param_env).may_apply()
```

Comment:
Please leave a comment that this is only going to be *ambiguous* if there are incoherent impls, because otherwise ambiguity should never happen in MIR.